mayastor/src/bdev/dev/iscsi.rs

```diff
@@ -52,24 +53,31 @@ impl TryFrom<&Url> for Iscsi {
             });
         }
 
-        if segments.len() > 1 {
+        if segments.len() > 2 {
             return Err(NexusBdevError::UriInvalid {
                 uri: url.to_string(),
                 message: String::from("too many path segments"),
             });
         }
 
-        let parameters: HashMap<String, String> =
+        let mut parameters: HashMap<String, String> =
             url.query_pairs().into_owned().collect();
 
+        let uuid = uri::uuid(parameters.remove("uuid"), url)?;
+
         if let Some(keys) = uri::keys(parameters) {
             warn!("ignored parameters: {}", keys);
         }
 
         Ok(Iscsi {
             name: url.to_string(),
             iqn: format!("{}:{}", ISCSI_IQN_PREFIX, Uuid::new_v4()),
-            url: format!("{}/0", url.to_string()),
+            url: if segments.len() == 2 {
+                url[.. url::Position::AfterPath].to_string()
+            } else {
+                format!("{}/0", &url[.. url::Position::AfterPath])
+            },
```

Comment:
I would be tempted to also "parse" the  LUN segment to make sure it's actually an integer.

Reply:
I didn't do it because spdk already does this for us:
[2020-07-24T09:27:10.836709918Z ERROR bdev_iscsi.c:824] could not parse URL: Invalid URL iscsi://127.0.0.1:3262/iqn.2019-05.io.openebs:replica1/blahnotanumber
Could not parse <lun>
iSCSI URL must be of the form: "iscsi://[<username>[%<password>]@]<host>[:<port>]/<target-iqn>/<lun>"